bsp/src/org/jetbrains/bsp/project/importing/BspResolverLogic.scala

```diff
@@ -415,8 +426,15 @@ private[importing] object BspResolverLogic {
       val nonEmptyParts = parts.filter(_.nonEmpty)
       if (nonEmptyParts.size > 1) nonEmptyParts.mkString("(", "+", ")") else nonEmptyParts.mkString
     }
-    head.map(combine).mkString +
+    val ret = head.map(combine).mkString +
       (if (tail.nonEmpty) tail.map(combine).mkString("(", "", ")") else tail.mkString)
+    if (ret.length > MaxModuleNameLength) {
+      val prefix = ret.substring(0, MaxModuleNameLength)
+      val suffix = md5(targets.map(_.getDisplayName).mkString)
```

Comment:
Just wondering if we can make this a little smarter so as to not have 2 very similar module names, or is it a rare enough occurrence not to matter in practice?

Reply:
that seems more complicated than worth it for this kind of edge case.  If you feel like adding that in later feel free to, but it's out of the scope of the amount of effort I'm willing to put into this fix.

Reply:
It's already better then it was when It didn't work at all